### PR TITLE
Feature/10 liquibase integration

### PR DESCRIPTION
**Pull request by damini1994**

### CODE REVIEW

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 1]
```diff
@@ -0,0 +1,182 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                                       http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.0.xsd">
+
+    <!-- Event Table -->
+    <changeSet id="1-create-event" author="damini">
+        <createTable tableName="event">
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: Lets follow table names to be plural like "events"

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: is nullable=false required? Primary Key is going to be unique and not-null by default

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="created_date" type="TIMESTAMP" />
+            <column name="last_modified_date" type="TIMESTAMP" />
+            <column name="created_by" type="VARCHAR(10)" />
+            <column name="last_modified_by" type="VARCHAR(10)" />
+            <column name="name" type="VARCHAR(255)" />
+            <column name="contact_number" type="VARCHAR(20)" />
+            <column name="email" type="VARCHAR(100)" />
+            <column name="address" type="VARCHAR(255)" />
+            <column name="website" type="VARCHAR(100)" />
+            <column name="status" type="VARCHAR(20)" />
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: can we add a check constraint here?

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="registration_date" type="VARCHAR(100)" />
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: type=timestamp

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    <changeSet id="2-create-vendor" author="damini">
+        <createTable tableName="vendor">
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="created_date" type="TIMESTAMP" />
+            <column name="last_modified_date" type="TIMESTAMP" />
+            <column name="created_by" type="VARCHAR(10)" />
+            <column name="last_modified_by" type="VARCHAR(10)" />
+            <column name="name" type="VARCHAR(255)" />
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: VARCHAR(20)

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="logo_url" type="VARCHAR(255)" />
+        </createTable>
+    </changeSet>
+
+    <!-- Venue Table -->
+    <changeSet id="3-create-venue" author="damini">
+        <createTable tableName="venue">
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml:68]: same as above

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="last_modified_date" type="TIMESTAMP" />
+            <column name="created_by" type="VARCHAR(10)" />
+            <column name="last_modified_by" type="VARCHAR(10)" />
+            <column name="name" type="VARCHAR(255)" />
+            <column name="address" type="VARCHAR(255)" />
+            <column name="latitude" type="DECIMAL(10, 6)" />
+            <column name="longitude" type="DECIMAL(10, 6)" />
+            <column name="total_seating_capacity" type="INTEGER" />
+            <column name="venue_type" type="VARCHAR(20)" />
+            <column name="operational_status" type="VARCHAR(20)" />
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: check constraint

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+        </createTable>
+    </changeSet>
+
+    <!-- EventShow Table -->
+    <changeSet id="4-create-event-show" author="damini">
+        <createTable tableName="event_show">
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml:92]: "event_shows"
same as above

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+        </createTable>
+    </changeSet>
+
+    <!-- EventType Table -->
+    <changeSet id="5-create-event-type" author="damini">
+        <createTable tableName="event_type">
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml:110]: event_types

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="last_modified_date" type="TIMESTAMP" />
+            <column name="created_by" type="VARCHAR(10)" />
+            <column name="last_modified_by" type="VARCHAR(10)" />
+            <column name="label" type="VARCHAR(15)" />
+        </createTable>
+    </changeSet>
+
+    <!-- Seat Table -->
+    <changeSet id="6-create-seat" author="damini">
+        <createTable tableName="seat">
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: seats

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="created_date" type="TIMESTAMP" />
+            <column name="last_modified_date" type="TIMESTAMP" />
+            <column name="created_by" type="VARCHAR(10)" />
+            <column name="last_modified_by" type="VARCHAR(10)" />
+            <column name="seat_type_id" type="CHAR(36)" />
+            <column name="venue_id" type="CHAR(36)" />
+            <column name="operational_status" type="VARCHAR(20)" />
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: check constraint

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="label" type="VARCHAR(5)" />
+        </createTable>
+    </changeSet>
+
+    <!-- SeatType Table -->
+    <changeSet id="7-create-seat-type" author="damini">
+        <createTable tableName="seat_type">
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: seat_types

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+        <createTable tableName="seat_type_in_show">
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="created_date" type="TIMESTAMP" />
+            <column name="last_modified_date" type="TIMESTAMP" />
+            <column name="created_by" type="VARCHAR(10)" />
+            <column name="last_modified_by" type="VARCHAR(10)" />
+            <column name="seat_type_id" type="CHAR(36)" />
+            <column name="show_id" type="CHAR(36)" />
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: nullable = false

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+    <changeSet id="9-create-seat-type-in-show" author="damini">
+        <createTable tableName="seat_type_in_show">
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="created_date" type="TIMESTAMP" />
+            <column name="last_modified_date" type="TIMESTAMP" />
+            <column name="created_by" type="VARCHAR(10)" />
+            <column name="last_modified_by" type="VARCHAR(10)" />
+            <column name="seat_type_id" type="CHAR(36)" />
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: nullable=false

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="booking_status" type="VARCHAR(20)" />
+        </createTable>
+    </changeSet>
+
+    <!-- SeatTypeInShow Table -->
+    <changeSet id="9-create-seat-type-in-show" author="damini">
+        <createTable tableName="seat_type_in_show">
+            <column name="id" type="CHAR(36)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml:176]: seat_type_in_shows

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 16]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="price" type="DECIMAL(10, 2)" />
+        </createTable>
+    </changeSet>
+
+    <changeSet id="10-create-show-genre" author="damini">
+        <createTable tableName="show_genre">
+            <column name="show_id" type="CHAR(36)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="genre" type="VARCHAR(15)">
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: change name to label

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+            <column name="show_id" type="CHAR(36)" />
+            <column name="price" type="DECIMAL(10, 2)" />
+        </createTable>
+    </changeSet>
+
+    <changeSet id="10-create-show-genre" author="damini">
+        <createTable tableName="show_genre">
+            <column name="show_id" type="CHAR(36)">
+                <constraints nullable="false"/>
+            </column>
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml:195]: same as above

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/add-constraints.xml [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+    </changeSet>
+
+
+    <changeSet id="2-add-not-null-constraint-event-title" author="damini">
+        <addNotNullConstraint tableName="event" columnName="title" columnDataType="VARCHAR(255)" />
+    </changeSet>
+
+    <changeSet id="3-add-not-null-constraint-seat-type" author="damini">
+        <addNotNullConstraint tableName="seat" columnName="seat_type_id" columnDataType="CHAR(36)" />
+    </changeSet>
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/add-constraints.xml]: we can directly add the not_null constraint while creating table

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/add-constraints.xml [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+
+    <!-- Primary Key Constraints -->
+    <changeSet id="1-add-primary-key-show-genre" author="damini">
+        <addPrimaryKey tableName="show_genre" columnNames="show_id, genre"/>
+    </changeSet>
+
+    <!-- Additional Constraints (Nullable, Unique, etc.) -->
+    <changeSet id="1-add-unique-constraint-venue-name" author="damini">
+        <addUniqueConstraint tableName="venue" columnNames="name" constraintName="uk_venue_name" />
+    </changeSet>
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/add-constraints.xml]: add constraint while creating table

**Review by rohitnandi12 — CHANGES_REQUESTED** [review]
Add this in properties file

spring.jpa.hibernate.ddl-auto: validate

run the application and update the Entities if it is breaking

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml (hunk 3 above)
Comment (damini1994) [reply book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: Yes, we can
Added check constraint

File: book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml [hunk 20]
```diff
@@ -0,0 +1,202 @@
...
+                <constraints checkConstraint="status IN ('ACTIVE', 'INACTIVE', 'SUSPENDED')"/>
+            </column>
+            <column name="registration_date" type="TIMESTAMP" />
+            <column name="logo_url" type="VARCHAR(255)" />
+        </createTable>
+    </changeSet>
+
+    <!-- Venue Table -->
+    <changeSet id="3-create-venue" author="damini">
+        <createTable tableName="venue">
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/resources/db/changelog/v1.0/create-tables.xml]: "venues"

File: book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/common/entity/Venue.java [hunk 21]
```diff
@@ -25,6 +26,7 @@
 @NoArgsConstructor
 public class Venue extends BaseAuditingEntity {
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/common/entity/Venue.java:29]: @ Table(venues)

File: book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/common/entity/Vendor.java [hunk 22]
```diff
@@ -1,35 +1,57 @@
...
     @Enumerated(EnumType.STRING)
+    @Column(nullable = false, length = 20)
+    @NotNull(message = "Status cannot be null.")
     private VendorStatus status;
 
-    @Column(nullable = true, length = 100)
-    @Size(min = 10, max = 100, message = "Registration date must be between 10 and 100 characters.")
-    private String registrationDate;
+    @Column(nullable = false)
+    @NotNull(message = "Registration date cannot be null.")
```
Comment (rohitnandi12) [inline book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/common/entity/Vendor.java]: @NotNull or @NotBlank are spring validations and not hibernate validations. They are valid only in DTO/Reques objects.

Remove all Spring Validations at Entity level.

**Review by rohitnandi12 — CHANGES_REQUESTED** [review]

**Review by rohitnandi12 — APPROVED** [review]

**Pull request merged**